dashboards-observability/public/components/visualizations/charts/box_plot/box_plot.tsx

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { isEmpty, last } from 'lodash';
+import { Plt } from '../../plotly/plot';
+import { EmptyPlaceholder } from '../../../event_analytics/explorer/visualizations/shared_components/empty_placeholder';
+import { AvailabilityUnitType } from '../../../event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_availability';
+import { ThresholdUnitType } from '../../../event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_thresholds';
+import { ConfigListEntry } from '../../../../../common/types/explorer';
+import { hexToRgb, filterDataConfigParameter } from '../../../event_analytics/utils/utils';
+import { FILLOPACITY_DIV_FACTOR } from '../../../../../common/constants/shared';
+import { LONG_CHART_COLOR, PLOTLY_COLOR } from '../../../../../common/constants/shared';
+import { DefaultBoxChartStyles } from '../../../../../common/constants/explorer';
+
+const { PointPosition } = DefaultBoxChartStyles;
+export const BoxPlot = ({ visualizations, layout, config }: any) => {
+  const DEFAULT_LABEL_SIZE = 10;
+  const { vis } = visualizations;
+  const {
+    data,
+    metadata: { fields },
+  } = visualizations.data.rawVizData;
+  const lastIndex = fields.length - 1;
+  const {
+    dataConfig = {},
+    layoutConfig = {},
+    availabilityConfig = {},
+  } = visualizations?.data?.userConfigs;
+  const valueForXSeries = dataConfig?.valueOptions?.dimensions
+    ? filterDataConfigParameter(dataConfig.valueOptions.dimensions)
+    : [];
+  const valueSeries = dataConfig?.valueOptions?.metrics
+    ? filterDataConfigParameter(dataConfig.valueOptions.metrics)
+    : [];
+  const boxOrientation = dataConfig?.chartStyles?.orientation || vis.orientation;
+  const isVertical = boxOrientation === vis.orientation;
+  const boxMode = dataConfig?.chartStyles?.boxMode || vis.boxmode;
+  let box;
+
+  if (isEmpty(valueSeries) || (boxMode === 'group' && valueForXSeries.length === 0)) {
```

Comment:
can we check this condition one more time `(boxMode === 'group' && valueForXSeries.length === 0)`

Reply:
in case of group mode, if there is no dimensions then empty plot is drawn so empty placeholder is return in that case